pyccl/background.py

```diff
@@ -112,6 +114,38 @@ def comoving_angular_distance(cosmo, a):
                          lib.comoving_angular_distance_vec, cosmo, a)
 
 
+def angular_diameter_distance(cosmo, a1, a2=None):
+    """Angular diameter distance.
+
+   .. note:: The angular diameter distance in Mpc from scale factor
+    a1 to scale factor a2. If a2 is not provided, it is assumed that
+    the distance will be calculated between 1 and a1.
+    This routine correspondes to Eq. (19) of astro-ph/9905116.
```

Comment:
Maybe specify that a2 should be <= a1? 
Also that, if they are both arrays, they should have the same size.
I'd also remove the reference to 9905116, since I think what we have is more general than that, right?